src/core.c/CompUnit/Repository/FileSystem.pm6

```diff
@@ -57,12 +57,13 @@ class CompUnit::Repository::FileSystem
     method id() {
         ⚛$!id // cas $!id, {
             $_ // do with self!distribution -> $distribution {
+                my $meta6 := Rakudo::Internals::JSON.to-json: :sorted-keys, $distribution.meta.hash;
                 my $parts :=
                     grep { .defined }, (.id with self.next-repo), slip # slip next repo id into hash parts to be hashed together
                     map  { nqp::sha1($_) },
                     map  { $distribution.content($_).open(:enc<iso-8859-1>).slurp(:close) },
                     $distribution.meta<provides>.values.unique.sort;
```

Comment:
Does it still makes sense having this line since the entire META6 is taken into account?

Reply:
Yes, because the contents of those files can change.